### ISSUE TITLE
GC: Restore: SharePoint: List Complication

### ISSUE THREAD

**Issue by dadams39** [alcionai/corso#1978]
- Restored List goes directly to `Recycle Bin` in Site Contents
- None of the column information is present either.

**Comment by dadams39** [comment]
- Inspection of the stored bytes show all items are present. However, identifiable information is preventing the restore of the item correctly.
- Lists are identified by the webURL. A restored List currently shares the same web URL as its predecessor. The remedy is to use the support package as in a similar manner as we used for messages, contacts and events.

**Comment by dadams39** [comment]
- Verified ContentType and Template are correct.
- Received error column definitions:
```bash
invalidRequest Unable to determine type of provided column definition
```
Investigation shows that Column `Title` is present at creation of new List.

List Items cannot be restored without the column structure.

**Comment by dadams39** [comment]
Troubleshooting Progress
----
Columns have a field `columnGroup`. When a user creates a column for their list, the columns from their list are added to the group, "Custom Columns".  Function required in support to decipher between legacy column names and user-generated ones. The litmus test is that legacy columns do not have enough information to be parsed through the REST API. See error above.

**Comment by dadams39** [comment]
- Verification of legacy columns completed.  #2012 will add functionality into code base
- List Items also have certain fields that are `read-only` and cannot be restored. #2011 will address these issues.
After these PRs are merged, #1964 will establish the restore pipeline.
